promlog/log.go

```diff
@@ -111,13 +111,16 @@ type Config struct {
 // New returns a new leveled oklog logger. Each logged line will be annotated
 // with a timestamp. The output always goes to stderr.
 func New(config *Config) log.Logger {
-	var l log.Logger
 	if config.Format != nil && config.Format.s == "json" {
-		l = log.NewJSONLogger(log.NewSyncWriter(os.Stderr))
```

Comment:
Why is it better not to use `WithSyncWriter`?

Reply:
This changes is out of scope of this PR. I revert this. Sorry for the confusion here.